includes/core/DataTransferObjects/API/class-response.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * Response DTO.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\DataTransferObjects;
+
+use WCPay\Core\Contracts\API\Response\Base_Response;
+use WP_Http_Cookie;
+
+/**
+ * DTO for parsing response from WCPay server API.
+ */
+final class Response extends Data_Transfer_Object implements Base_Response {
+	/**
+	 * Response body.
+	 *
+	 * @var string $body
+	 */
+	private $body;
+
+	/**
+	 * Status code.
+	 *
+	 * @var int $code
+	 */
+	private $code;
+
+	/**
+	 * Response headers.
+	 *
+	 * @var array $headers
+	 */
+	private $headers;
+
+	/**
+	 * Response message.
+	 *
+	 * @var string $message
+	 */
+	private $message;
+
+	/**
+	 * Response cookies.
+	 *
+	 * @var WP_Http_Cookie[] $cookies
+	 */
+	private $cookies;
+
+	/**
+	 * Parsed response array.
+	 *
+	 * @var array $response_dody
+	 */
+	private $response_data;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param string           $body Response body.
+	 * @param int              $code Response code.
+	 * @param array            $headers Response headers.
+	 * @param string           $message Response message.
+	 * @param WP_Http_Cookie[] $cookies Response cookies.
+	 */
+	public function __construct( $body, $code, $headers, $message, $cookies ) {
+		$this->body          = $body;
+		$this->code          = $code;
+		$this->headers       = $headers;
+		$this->message       = $message;
+		$this->cookies       = $cookies;
+		$this->response_data = json_decode( $body, true );
+	}
+
+	/**
+	 * Returns true if response is JSON.
+	 *
+	 * @return bool
+	 */
+	public function is_json_response() {
+		return 'application/json' === substr( $this->headers['content-type'] ?? '', 0, strlen( 'application/json' ) );
+	}
+
+	/**
+	 * Get body.
+	 *
+	 * @return string
+	 */
+	public function get_body() {
```

Comment:
`$body` and `get_body()` represents raw response from the server, and `get_response_data` is `json_decoded` version of `get_body`. I had an idea that you might need raw data from the server sometimes, and that's why I kept it.

Reply:
I don't think we have any cases where the raw response is currently being used. Let's remove this method in order to enforce proper implementation for edge cases. What do you think? I don't have any realistic scenarios in my mind, but in case it is needed, maybe it will be better to keep whatever it is needed for in the response class 🤔

Reply:
I will remove get body. We can easily return it if needed.